src/Generator/InvoicePdfFileGenerator.php

```diff
@@ -38,30 +31,22 @@ final class InvoicePdfFileGenerator implements InvoicePdfFileGeneratorInterface
     private $invoiceLogoPath;
 
     public function __construct(
-        InvoiceRepository $invoiceRepository,
-        ChannelRepositoryInterface $channelRepository,
         EngineInterface $templatingEngine,
         GeneratorInterface $pdfGenerator,
         FileLocatorInterface $fileLocator,
         string $template,
         string $invoiceLogoPath
     ) {
-        $this->invoiceRepository = $invoiceRepository;
-        $this->channelRepository = $channelRepository;
         $this->templatingEngine = $templatingEngine;
         $this->pdfGenerator = $pdfGenerator;
         $this->fileLocator = $fileLocator;
         $this->template = $template;
         $this->invoiceLogoPath = $invoiceLogoPath;
     }
 
-    public function generate(string $invoiceId): InvoicePdf
+    public function generate(InvoiceInterface $invoice, ChannelInterface $channel): InvoicePdf
```

Comment:
Should we pass a channel to the `generate` function? What if we pass a different channel than the invoice's channel? I think we should still get the channel from the repository basing on data from the invoice, as it would be set even if the invoice would not be persisted to the database.